test/e2e/Deploy.t.sol

```diff
@@ -85,12 +167,168 @@ abstract contract CommonDeploymentTest is HaiTest, Deploy {
+    ParamChecker._checkParams(address(stabilityFeeTreasury), abi.encode(_stabilityFeeTreasuryParams));
+  }
+
+  // GlobalSettlement
+  function test_GlobalSettlement_Bytecode() public {
+    assertEq(address(globalSettlement).code, type(GlobalSettlement).runtimeCode);
+  }
+
+  function test_GlobalSettlement_Auth() public {
+    // only governor
+    assertEq(globalSettlement.authorizedAccounts().length, _governorAccounts);
+  }
+
+  function test_GlobalSettlement_Params() public view {
+    ParamChecker._checkParams(address(globalSettlement), abi.encode(_globalSettlementParams));
+  }
+
+  // PostSettlementSurplusAuctionHouse
+  function test_PostSettlementSurplusAuctionHouse_Bytecode() public {
+    assertEq(address(postSettlementSurplusAuctionHouse).code, type(PostSettlementSurplusAuctionHouse).runtimeCode);
+  }
+
+  function test_PostSettlementSurplusAuctionHouse_Auth() public {
+    assertEq(postSettlementSurplusAuctionHouse.authorizedAccounts(address(settlementSurplusAuctioneer)), true);
+
+    // 1 contract + governor accounts
+    assertEq(postSettlementSurplusAuctionHouse.authorizedAccounts().length, 1 + _governorAccounts);
+  }
+
+  function test_PostSettlementSurplusAuctionHouse_Params() public view {
+    ParamChecker._checkParams(address(postSettlementSurplusAuctionHouse), abi.encode(_postSettlementSAHParams));
+  }
+
+  // PostSettlementAuctioneer
+  function test_PostSettlementAuctioneer_Bytecode() public {
+    assertEq(address(settlementSurplusAuctioneer).code, type(SettlementSurplusAuctioneer).runtimeCode);
+  }
+
+  function test_PostSettlementAuctioneer_Auth() public {
+    // only governor
+    assertEq(settlementSurplusAuctioneer.authorizedAccounts().length, _governorAccounts);
+  }
+
+  // Governance checks
   function test_Grant_Auth() public {
     _test_Authorizations(governor, true);
     if (delegate != address(0)) _test_Authorizations(delegate, true);
     _test_Authorizations(deployer, false);
   }
 
+  function test_Timelock_Bytecode() public {
+    assertEq(address(timelock).code, type(TimelockController).runtimeCode);
+  }
+
+  function test_Timelock_Auth() public {
+    assertEq(timelock.hasRole(keccak256('PROPOSER_ROLE'), address(haiGovernor)), true);
+    assertEq(timelock.hasRole(keccak256('CANCELLER_ROLE'), address(haiGovernor)), true);
+    assertEq(timelock.hasRole(keccak256('EXECUTOR_ROLE'), address(0)), true);
+  }
+
+  function test_Timelock_Params() public {
+    assertEq(timelock.getMinDelay(), _governorParams.timelockMinDelay);
+  }
+
+  function test_HaiGovernor_Bytecode_MANUAL_CHECK() public {
+    // Not possible to check bytecode because it has immutable storage
+    // Needs to be manually checked
+  }
+
+  function test_HaiGovernor_Params() public {
+    assertEq(haiGovernor.votingDelay(), _governorParams.votingDelay);
+    assertEq(haiGovernor.votingPeriod(), _governorParams.votingPeriod);
+    assertEq(haiGovernor.proposalThreshold(), _governorParams.proposalThreshold);
```

Comment:
Consider using `ParamChecker._checkParams()` for the newly introduced contracts.

Reply:
hmm but doesn't have a `params` struct, not sure if it's worth to do so, since is not even modifiable

Reply:
> hmm but doesn't have a params struct

Wdym? `HaiGovernorParams` and `TokenDistributorParams` structs exist within `IHaiGovernor` and `ITokenDistributor`, respectively.

> not sure if it's worth to do so, since is not even modifiable

I agree that it may not be worth adding a `params()` method to them.